### PR TITLE
User cannot talk to themselves and greetings check added

### PR DESCRIPTION
**Pull request by kevinzhangftw**

### CODE REVIEW

File: src/sockets/websocket.py [hunk 1]
```diff
@@ -111,10 +111,18 @@ def handleNameStagingState(self, user, str):
     def handleReadyState(self, user, str):
         #check existence of name and message
         if not ProcessText.hasNameandMessage(str):
-            self.write_message("who is the recipient and what is the message")
+            if ProcessText.hasGreetings(str):
```
Comment (jyleong) [inline src/sockets/websocket.py]: can be simplified to 
not ProcessText.hasNameandMessage(str) and ProcessText.hasGreetings(str)

File: src/sockets/websocket.py [hunk 2]
```diff
@@ -129,6 +137,12 @@ def handleReadyState(self, user, str):
             self.countDown.start()
             recipient.socket.countDown.start()
 
+    def verifyRecipientName(self, user, possibleRecipientName):
+        if user.name == possibleRecipientName:
```
Comment (jyleong) [inline src/sockets/websocket.py]: simplify to:

```return not (user.name == possibleRecipientName)```

File: src/textProcessing/ProcessText.py [hunk 3]
```diff
@@ -69,5 +76,14 @@ def hasNameandMessage(userInput):
 
     @staticmethod
     def hasRecipientName(userInput):
-        targetPhrase = ['yo', 'yah', 'hey']
+        targetPhrase = ['yo', 'yah', 'hey', 'hello']
         return any (phrase in userInput.split() for phrase in targetPhrase)
+
+    @staticmethod
+    def hasGreetings(userInput):
+        targetPhrase = ['yo', 'yah', 'hey', 'hello']
```
Comment (kimjune01) [inline src/textProcessing/ProcessText.py]: missing 'hi', 'howdy', 'greetings'?

File: src/textProcessing/ProcessText.py [hunk 4]
```diff
@@ -69,5 +76,14 @@ def hasNameandMessage(userInput):
...
+        for phrase in targetPhrase:
+            if userInput.split()[0] == phrase:
+                return True
+            else:
```
Comment (jyleong) [inline src/textProcessing/ProcessText.py]: I think the false clause has to be moved out of the for loop scope

**Review by jyleong — CHANGES_REQUESTED** [review]
I'll need you to make these changes that I listed above

File: src/sockets/websocket.py (hunk 1 above)
Comment (kimjune01) [reply src/sockets/websocket.py]: at that point, it should be extracted to a named function

**Review by jyleong — APPROVED** [review]

**Pull request merged**